sdk/core/src/date.rs

```diff
@@ -0,0 +1,185 @@
+//! Azure date and time parsing and formatting
+
+// RFC 3339 vs ISO 8601
+// https://ijmacd.github.io/rfc3339-iso8601/
+
+use crate::error::{ErrorKind, ResultExt};
+use time::{
+    format_description::{
+        well_known::{Rfc2822, Rfc3339},
+        FormatItem,
+    },
+    macros::format_description,
+    OffsetDateTime, PrimitiveDateTime, UtcOffset,
+};
+
+/// RFC 2822: Internet Message Format
+///
+/// https://www.rfc-editor.org/rfc/rfc2822#section-3.3
+/// https://www.rfc-editor.org/rfc/rfc2822#section-4.3
+/// A "GMT" zone is obsolete, instead "+0000" is used.
+///
+/// Date: Fri, 21 Nov 1997 10:01:10 -0600
+pub fn parse_rfc2822(s: &str) -> crate::Result<OffsetDateTime> {
+    OffsetDateTime::parse(s, &Rfc2822).with_context(ErrorKind::DataConversion, || {
+        format!("unable to parse smtp date '{s}")
+    })
+}
+
+/// RFC 2822: Internet Message Format
+///
+/// https://www.rfc-editor.org/rfc/rfc2822#section-3.3
+/// https://www.rfc-editor.org/rfc/rfc2822#section-4.3
+/// A "GMT" zone is obsolete, instead "+0000" is used.
+///
+/// Date: Fri, 21 Nov 1997 10:01:10 -0600
+pub fn to_rfc2822(date: &OffsetDateTime) -> String {
+    // known format does not panic
+    date.format(&Rfc2822).unwrap()
+}
+
+/// RFC 3339: Date and Time on the Internet: Timestamps
+///
+/// https://www.rfc-editor.org/rfc/rfc3339
+///
+/// 1985-04-12T23:20:50.52Z
+pub fn parse_rfc3339(s: &str) -> crate::Result<OffsetDateTime> {
+    OffsetDateTime::parse(s, &Rfc3339).with_context(ErrorKind::DataConversion, || {
+        format!("unable to parse internet date '{s}")
```

Comment:
I think these error messages should reflect the same nomenclature used in the method names.

Reply:
Me too. 82c63f8c6166da225e11092fc0091c79e48868ca had already changed it.